tests/support/mod.rs

```diff
@@ -396,3 +396,33 @@ impl<T> Sink01 for DeadSink<T> {
         Ok(futures01::Async::Ready(()))
     }
 }
+
+/// Takes a test name and a future, and uses `rusty_fork` to perform a cross-platform
+/// process fork. This allows us to test functionality without conflicting with global
+/// state that may have been set/mutated from previous tests
+pub fn fork_test<T: std::future::Future<Output = ()>>(test_name: &'static str, fut: T) {
+    let fork_id = rusty_fork::rusty_fork_id!();
+
+    rusty_fork::fork(
+        test_name,
+        fork_id,
+        |_| {},
+        |child, f| {
+            let status = child.wait().expect("Couldn't wait for child process");
```

Comment:
rusty_fork provides a [`wait_with_output`](https://docs.rs/rusty-fork/0.3.0/rusty_fork/struct.ChildWrapper.html#method.wait_with_output) method, which provides a direct buffer to stdout via its `.stdout`, sidestepping the need to copy from the captured `File`.

However, child is shared as `&mut child` to this closure, and `wait_with_output` returns an owned value, so it wasn't clear how this could be used.